EventFiltering/PWGHF/HFFilterHelpers.h

```diff
@@ -634,38 +674,42 @@ inline bool HfFilterHelper::isSelectedProton4Femto(const T1& track, const T2& tr
     return false; // use only global tracks
   }
 
-  float NSigmaTPC = track.tpcNSigmaPr();
-  float NSigmaTOF = track.tofNSigmaPr();
+  // PID evaluation
+  float NSigmaTPC = (trackSpecies == kProtonForFemto) ? track.tpcNSigmaPr() : track.tpcNSigmaDe();
+  float NSigmaTOF = (trackSpecies == kProtonForFemto) ? track.tofNSigmaPr() : track.tofNSigmaDe();
   if (!forceTof && !track.hasTOF()) {
     NSigmaTOF = 0.; // always accepted
   }
 
+  // mTpcPidCalibrationOption only available for proton, dummy for deuteron
   if (mTpcPidCalibrationOption == 1) {
-    NSigmaTPC = getTPCPostCalib(track, kPr);
+    NSigmaTPC = getTPCPostCalib(track, trackSpecies == kProtonForFemto ? kPr : kDe);
   } else if (mTpcPidCalibrationOption == 2) {
     if (track.sign() > 0) {
-      NSigmaTPC = getTPCSplineCalib(track, kPr);
+      NSigmaTPC = getTPCSplineCalib(track, trackSpecies == kProtonForFemto ? kPr : kDe);
     } else {
-      NSigmaTPC = getTPCSplineCalib(track, kAntiPr);
+      NSigmaTPC = getTPCSplineCalib(track, trackSpecies == kProtonForFemto ? kAntiPr : kAntiDe);
     }
   }
 
   float NSigma = std::sqrt(NSigmaTPC * NSigmaTPC + NSigmaTOF * NSigmaTOF);
 
+  // Apply particle-specific PID cuts
   if (trackPar.getPt() <= mPtThresholdPidStrategyForFemto) {
```

Comment:
I would consider to have also `mPtThresholdPidStrategyForFemto` species dependent (in particular we might want to use only TPC for the deuteron at low pT), so we might use even a different strategy for the deuteron (TPC and TOF above this threshold, TPC only below)

Reply:
yes, I adjusted the logic for this purpose, you can have a look to make sure it's correct